### PR TITLE
Update Runtime Provisioner documentation

### PR DESCRIPTION
**Pull request by majakurcius**
**Description**

Changes proposed in this pull request:

- Update the Runtime Provisioner documentation after introducing the following functionalities:
  * The Runtime Provisioner registers Runtimes in the Director.
  * The Runtime Provisioner accepts a list of components to install on the provisioned Runtime along with their custom configuration.
  * The Runtime Provisioner accepts a custom configuration for the provisioned Runtime.
  * Successful deprovisioning cleans up Runtime data.

**Related issue(s)**
#454, #399, #477

### CODE REVIEW

File: docs/provisioner/01-01-provisioner-overview.md [hunk 1]
```diff
@@ -3,19 +3,15 @@ title: Overview
 type: Overview
 ---
 
-The Runtime Provisioner is a Compass component responsible for provisioning, installing, and deprovisioning clusters with Kyma (Kyma Runtimes). The relationship between clusters and Runtimes is 1:1.
+The Runtime Provisioner is a Compass component responsible for provisioning, installing, and deprovisioning clusters with Kyma (Kyma Runtimes). The relationship between clusters and Runtimes is 1:1. The Runtime Provisioner is registered in the Director as an integration system.
```
Comment (klaudiagrz) [inline docs/provisioner/01-01-provisioner-overview.md]: `Runtime Provisioner` vs. `Provisioner`? We need more consistency throughout the Compass docs. Also: `Director` vs. `the Director`?

File: docs/provisioner/01-01-provisioner-overview.md [hunk 2]
```diff
@@ -3,19 +3,15 @@ title: Overview
...
     * GCP
     * Microsoft Azure
     * Amazon Web Services (AWS).
     
-Note that the operations of provisioning and deprovisioning are asynchronous. They return the operation ID, which you can use to [check the Runtime Operation Status](08-03-runtime-operation-status.md).
-
-The Runtime Provisioner also allows you to [clean up Runtime data](08-06-clean-up-runtime-data.md). This operation removes the data for a given Runtime from the database and frees up the Runtime ID for reuse. It is useful when your cluster has died or when the operation of deprovisioning has failed.
-
-> **CAUTION:** The operation of cleaning up Runtime data does not trigger Runtime deprovisioning and the cluster might still exist after running it.
+Note that the operations of provisioning and deprovisioning are asynchronous. The operation of provisioning returns the Runtime Operation Status containing the runtime ID and the operation ID. The operation of deprovisioning returns the operation ID. You can use the operation ID to [check the Runtime Operation Status](08-03-runtime-operation-status.md).
```
Comment (klaudiagrz) [inline docs/provisioner/01-01-provisioner-overview.md]:
```suggestion
Note that the operations of provisioning and deprovisioning are asynchronous. Provisioning returns the Runtime Operation Status that contains the Runtime ID and the operation ID. Deprovisioning returns the operation ID. You can use the operation ID to [check the Runtime Operation Status](08-03-runtime-operation-status.md).
```

Comment (klaudiagrz) [reply docs/provisioner/01-01-provisioner-overview.md]: Consider putting that info in a **NOTE**.

File: docs/provisioner/01-01-provisioner-overview.md [hunk 3]
```diff
@@ -26,3 +22,5 @@ To access the Runtime Provisioner, forward the port that the GraphQL Server is l
 ```bash
 kubectl -n compass-system port-forward svc/compass-provisioner 3000:3000
 ```
+
+When making a call to the Runtime Provisioner, make sure to attach a Tenant header to the request.
```
Comment (klaudiagrz) [inline docs/provisioner/01-01-provisioner-overview.md]:
```suggestion
When making a call to the Runtime Provisioner, make sure to attach a tenant header to the request.
```

Comment (klaudiagrz) [reply docs/provisioner/01-01-provisioner-overview.md]: Why the capital letter for tenant? Also: `Runtime Provisioner` vs. `Provisioner`?

File: docs/provisioner/08-02-provisioning-gardener.md [hunk 4]
```diff
@@ -73,17 +73,24 @@ This tutorial shows how to provision clusters with Kyma Runtimes on Google Cloud
 
   4. Download the service account configuration (`kubeconfig.yaml`) and use it to create a Secret in the `compass-system` Namespace with the key named `credentials` and the value encoded with base64.
 
-  5. Make a call to the Runtime Provisioner to create a cluster on GCP.
+     ```bash
+     kubectl -n compass-system create secret generic {SECRET_NAME} --from-file=credentials={PATH_TO_KUBECONFIG}
+     ```
     
-      > **NOTE:** The cluster name must start with a lowercase letter followed by up to 19 lowercase letters, numbers, or hyphens, and cannot end with a hyphen.                                                                 
+  5. Make a call to the Runtime Provisioner with a **tenant** header to create a cluster on GCP. Note, that the Runtime Agent component (`compass-runtime-agent`) in the Kyma configuration is mandatory, and that the order of the components matters.
```
Comment (klaudiagrz) [inline docs/provisioner/08-02-provisioning-gardener.md]:
```suggestion
  5. Make a call to the Runtime Provisioner with a **tenant** header to create a cluster on GCP. Note that the Runtime Agent component (`compass-runtime-agent`) in the Kyma configuration is mandatory and that the order of the components matters.
```

File: docs/provisioner/08-02-provisioning-gardener.md [hunk 5]
```diff
@@ -137,19 +170,26 @@ This tutorial shows how to provision clusters with Kyma Runtimes on Google Cloud
 
   4. Download the service account configuration (`kubeconfig.yaml`) and use it to create a Secret in the `compass-system` Namespace with the key `credentials` and the value encoded with base64.
 
-  5. Make a call to the Runtime Provisioner to create a cluster on Azure.
+     ```bash
+     kubectl -n compass-system create secret generic {SECRET_NAME} --from-file=credentials={PATH_TO_KUBECONFIG}
+     ```
+    
+  5. Make a call to the Runtime Provisioner with a **tenant** header to create a cluster on Azure. Note, that the Runtime Agent component (`compass-runtime-agent`) in the Kyma configuration is mandatory, and that the order of the components matters.
```
Comment (klaudiagrz) [inline docs/provisioner/08-02-provisioning-gardener.md]:
```suggestion
  5. Make a call to the Runtime Provisioner with a **tenant** header to create a cluster on Azure. Note that the Runtime Agent component (`compass-runtime-agent`) in the Kyma configuration is mandatory and that the order of the components matters.
```

File: docs/provisioner/08-02-provisioning-gardener.md [hunk 6]
```diff
@@ -201,20 +269,26 @@ This tutorial shows how to provision clusters with Kyma Runtimes on Google Cloud
 
   4. Download the service account configuration (`kubeconfig.yaml`) and use it to create a Secret in the `compass-system` Namespace with the key `credentials` and the value encoded with base64.
 
-  5. Make a call to the Runtime Provisioner to create a cluster on AWS.
+     ```bash
+     kubectl -n compass-system create secret generic {SECRET_NAME} --from-file=credentials={PATH_TO_KUBECONFIG}
+     ```
+
+  5. Make a call to the Runtime Provisioner with a **tenant** header to create a cluster on AWS. Note, that the Runtime Agent component (`compass-runtime-agent`) in the Kyma configuration is mandatory, and that the order of the components matters.
```
Comment (klaudiagrz) [inline docs/provisioner/08-02-provisioning-gardener.md]:
```suggestion
  5. Make a call to the Runtime Provisioner with a **tenant** header to create a cluster on AWS. Note that the Runtime Agent component (`compass-runtime-agent`) in the Kyma configuration is mandatory and that the order of the components matters.
```

File: docs/provisioner/08-02-provisioning-gardener.md [hunk 7]
```diff
@@ -251,12 +350,17 @@ This tutorial shows how to provision clusters with Kyma Runtimes on Google Cloud
...
         }
       }
       ```
   </details>
     
 </div>
 
-The operation of provisioning is asynchronous. Use the provisioning operation ID (`provisionRuntime`) to [check the Runtime Operation Status](08-03-runtime-operation-status.md) and verify that the provisioning was successful. Use the Runtime ID (`id`) to [check the Runtime Status](08-04-runtime-status.md). 
\ No newline at end of file
+The operation of provisioning is asynchronous. The operation of provisioning returns the Runtime Operation Status containing the runtime ID (`provisionRuntime.runtimeID`) and the operation ID (`provisionRuntime.id`). Use the Runtime ID to [check the Runtime Status](08-04-runtime-status.md). Use the provisioning operation ID to [check the Runtime Operation Status](08-03-runtime-operation-status.md) and verify that the provisioning was successful.
```
Comment (klaudiagrz) [inline docs/provisioner/08-02-provisioning-gardener.md]:
```suggestion
The operation of provisioning is asynchronous. Provisioning returns the Runtime Operation Status that contains the Runtime ID (`provisionRuntime.runtimeID`) and the operation ID (`provisionRuntime.id`). Use the Runtime ID to [check the Runtime Status](08-04-runtime-status.md). Use the provisioning operation ID to [check the Runtime Operation Status](08-03-runtime-operation-status.md) and verify that the provisioning is successful.
```

File: docs/provisioner/01-01-provisioner-overview.md (hunk 2 above)
Comment (majakurcius) [reply docs/provisioner/01-01-provisioner-overview.md]: I didn't cos there are already other notes and on top of looking ugly that would probably disappear amongst them.

File: docs/provisioner/08-02-provisioning-gardener.md [hunk 8]
```diff
@@ -101,24 +108,50 @@ This tutorial shows how to provision clusters with Kyma Runtimes on Google Cloud
...
+              ]
+              configuration: [
+                { 
+                  key: "{CONFIG_PROPERTY_KEY}"
+                  value: "{CONFIG_PROPERTY_VALUE}"
+                  secret: {true|false} # Specifies if the property is confidential
+                }
+              ]
             }
+            credentials: { secretName: "{GARDENER_SERVICE_ACCOUNT_CONFIGURATION_SECERT_NAME}" }
```
Comment (Maladie) [inline docs/provisioner/08-02-provisioning-gardener.md]: We should remove this part as credentials are now passed with envs

File: docs/provisioner/08-02-provisioning-gardener.md [hunk 9]
```diff
@@ -137,19 +170,26 @@ This tutorial shows how to provision clusters with Kyma Runtimes on Google Cloud
 
   4. Download the service account configuration (`kubeconfig.yaml`) and use it to create a Secret in the `compass-system` Namespace with the key `credentials` and the value encoded with base64.
```
Comment (Maladie) [inline docs/provisioner/08-02-provisioning-gardener.md]: We should move this part to requirements, as we expect that kubeconfig is already present during Provisioner installation

File: docs/provisioner/08-02-provisioning-gardener.md [hunk 10]
```diff
@@ -164,13 +204,38 @@ This tutorial shows how to provision clusters with Kyma Runtimes on Google Cloud
...
+              ]
+              configuration: [
+                { 
+                  key: "{CONFIG_PROPERTY_KEY}"
+                  value: "{CONFIG_PROPERTY_VALUE}"
+                  secret: {true|false} # Specifies if the property is confidential
+                }
+              ]
+            }
+            credentials: { secretName: "{GARDENER_SERVICE_ACCOUNT_CONFIGURATION_SECERT_NAME}" }
```
Comment (Maladie) [inline docs/provisioner/08-02-provisioning-gardener.md]: Please remove

File: docs/provisioner/08-02-provisioning-gardener.md [hunk 11]
```diff
@@ -231,18 +305,43 @@ This tutorial shows how to provision clusters with Kyma Runtimes on Google Cloud
...
```
Comment (Maladie) [inline docs/provisioner/08-02-provisioning-gardener.md]: Please remove this from the example call

File: docs/provisioner/01-01-provisioner-overview.md [hunk 12]
```diff
@@ -3,19 +3,17 @@ title: Overview
 type: Overview
 ---
 
-The Runtime Provisioner is a Compass component responsible for provisioning, installing, and deprovisioning clusters with Kyma (Kyma Runtimes). The relationship between clusters and Runtimes is 1:1.
+The Runtime Provisioner is a Compass component responsible for provisioning, installing, and deprovisioning clusters with Kyma (Kyma Runtimes). The relationship between clusters and Runtimes is 1:1. The Runtime Provisioner is registered in the Director as an Integration System. 
 
-It is powered by [Hydroform](https://github.com/kyma-incubator/hydroform) and it allows you to provision the clusters in the following ways:
+It allows you to provision the clusters in the following ways:
```
Comment (klaudiagrz) [inline docs/provisioner/01-01-provisioner-overview.md]:
```suggestion
It allows you to provision clusters in the following ways:
```

File: docs/provisioner/01-01-provisioner-overview.md [hunk 13]
```diff
@@ -3,19 +3,17 @@ title: Overview
...
     * GCP
     * Microsoft Azure
     * Amazon Web Services (AWS).
     
-Note that the operations of provisioning and deprovisioning are asynchronous. They return the operation ID, which you can use to [check the Runtime Operation Status](08-03-runtime-operation-status.md).
-
-The Runtime Provisioner also allows you to [clean up Runtime data](08-06-clean-up-runtime-data.md). This operation removes the data for a given Runtime from the database and frees up the Runtime ID for reuse. It is useful when your cluster has died or when the operation of deprovisioning has failed.
-
-> **CAUTION:** The operation of cleaning up Runtime data does not trigger Runtime deprovisioning and the cluster might still exist after running it.
+During the operation of provisioning, you can pass a list of Kyma components you want installed on the provisioned Runtime with their custom configuration, as well as a custom Runtime configuration.
```
Comment (klaudiagrz) [inline docs/provisioner/01-01-provisioner-overview.md]:
```suggestion
During the operation of provisioning, you can pass a list of Kyma components you want to install on the provisioned Runtime with their custom configuration, as well as a custom Runtime configuration. 
```

**Pull request merged**